Add .NET Framework to target frameworks

GerardSmit opened the pull request.

GerardSmit committed:
Add .NET Framework to target frameworks.

Nihlus commented on StrictEmit/StrictEmit.csproj:
```diff
@@ -46,6 +46,8 @@
     <PackageReference Include="JetBrains.Annotations" Version="11.1.0" PrivateAssets="all" />
     <PackageReference Include="JetBrainsAnnotations.Fody" Version="2.0.0" PrivateAssets="all" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
```
There's no need to conditionally exclude the package. The framework will resolve the correct dependecy at compile time.

GerardSmit replied:
The minium .NET Framework version for the package `System.Reflection.Emit` is 4.5, 
thus it gives an error if this is not excluded from .NET Framework 4.0.

Nihlus commented on StrictEmit/StrictEmit.csproj:
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net4</TargetFrameworks>
```
Can a lower version of .NET be targeted while we're at it?

GerardSmit replied:
.Net Framework: yes, we can go lower.
.Net Standard: needs code changes since .Net Standard 1.x has `Type.GetTypeInfo().IsValueType` instead of `Type.IsValueType`. This can be done with `#if`'s or an utility method.

GerardSmit committed:
Updated JetBrainsAnnotations.Fody

Also installed Fody 2.4.6. The older versions didn't support the .Net Core CLI.

GerardSmit commented:
Updated Fody because of issues with dotnet CLI. `dotnet build` wouldn't work with older version. This is fixed in newer releases.

GerardSmit committed:
Raised the .Net Framework version to 4.5

This allows us to remove the condition on the ItemGroup.

GerardSmit commented:
Increased .Net Framework to version 4.5 and removed the condition on the ItemGroup as discussed in Discord.
